firmware/shared/src/hw/hw_fdcan.c

```diff
@@ -1,11 +1,20 @@
-#include "io_can.h"
-#undef NDEBUG
+#include "hw_can.h"
+#undef NDEBUG // TODO remove this in favour of always_assert
 #include <assert.h>
 #include "io_log.h"
 #include "io_time.h"
+#include "io_canQueue.h"
 
-void io_can_init(const CanHandle *can_handle)
+/**
+ * @attention THIS MUST BE DEFINED IN YOUR CONFIGURATIONS
+ * @param hfdcan takes a handle to a STM32 HAL CAN object
+ * @returns a pointer to a CanHandle object (the metadata associated with the STM32 HAL CAN object)
+ */
+CanHandle *hw_can_getHandle(const FDCAN_HandleTypeDef *hfdcan);
```

Comment:
Maybe put this in the header file so its more clear?

Reply:
but then if you include "hw_can.h" then you will get this function? Is that desired?

Reply:
uhh i suppose not, maybe just comment then?